custom_components/micro_air_easytouch/climate.py

```
@@ -248,35 +248,53 @@ async def async_set_temperature(self, **kwargs: Any) -> None:
+            # Optimistic update
+            self._state["autoCool_sp"] = int(kwargs["target_temp_high"])
+            self._state["autoHeat_sp"] = int(kwargs["target_temp_low"])
 
         if changes:
+            # Optimistic update - show new state immediately
+            self.async_write_ha_state()
+            
             message = {"Type": "Change", "Changes": changes}
-            await self._data.send_command(self.hass, ble_device, message)
+            if not await self._data.send_command(self.hass, ble_device, message):
+                raise HomeAssistantError("Failed to set temperature")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)